lib/misc.coffee

```diff
@@ -1,181 +1,51 @@
     else
       @_model.val(prop, val)
 
+  # since we already assume that any instance of BaseClass uses a MapManager
+  # We can create it here, to save
+  # lines
+  # of
+  # code
+  _setModel: (@_model)->
+    if not @_model?
+      @_model = new Operation.MapManager(@).execute()
+      for n,v of @_tmp_model
+        @_model.val(n,v)
+
+  _setModel: (@_model)->
+    delete @_tmp_model
+
 
+###
 # Simple class that contains a word and links to the selections pointing
 # to it
 #
 class Word extends BaseClass
-    # Attribute containing the string
-
-  _name: "Word"
-
-  _getModel: (Y, Operation) ->
-    if @_model == null
-      @_model = new Operation.MapManager(@).execute()
-      # create left and right list of selections
-      left = new Operation.MapManager(@).execute()
-      left.insert 0, @left
-
-      right = new Operation.MapManager(@).execute()
-      right.insert 0, @right
-
-      # extend the lists
-      extend @_model, customList
-      extend left, customList
-      extend right, customList
-
-      @_model.val("left", left)
-      @_model.val("right", right)
-      @_model.val("word", @word)
-      @_model.val("richText", @richText)
-
-      @_setModel @_model
-    return @_model
-
-  _setModel: (model) ->
-    delete @left
-    delete @right
-    delete @word
-    delete @richText
-
-    @_model = model
-    @_model.observe = @observer
-
-  # Construct a new list of words
-  #
-  # @param [String] word The initial string value
-  # @return [Word] a word instance
-  #
-  constructor: (@word, @richText) ->
-    # Selections that have this word as left bound
-    @left = []
-    # Selections that have this word as right bound
-    @right = []
-
-  # Remove a selection from left or right array
-  #
-  # @param [Selection] selection the selection to remove
-  # @param [Option] side the side where to remove the selection
-  # @option side [String] left left side
-  # @option side [String] right right side
-  #
-  removeSel: (selection, side)->
-    if side == "left"
-      array = @_get("left")
-    else if side == "right"
-      array = @_get("right")
-    else
-      throw new Error "Invalid argument #{side}, expected 'left' or 'right'"
-    index = 0
-    for sel, index in array
-      if sel.equals selection
-        array.splice index, 1
-        break
-
-  # Get index of word in the list of words
-  #
-  # @return [Integer] the index of the word
-  index: ->
-    index = @richText.getWords(0).indexOf @
-    if index == -1
-      9e99
-    else
-      index
 
-  # @overload getSelections()
-  #   Return all the selections
-  #   @return [Array<Selection>] an array of selection
   #
-  # @overload getSelections(filter)
-  #   Return all the selections and filter using fun
-  #   @param [Function] filter the function to use for filtering
-  #   @return [Array<Selection>] an array of selection
+  ## TODO: explain what this does. Can I use it somewhere?
+  ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ###
```

Comment:
It gets the left and right selections bound to a word, concatenate the two arrays, filter them using the filter function and then return another array with unique entries.